### ISSUE TITLE
Emit custom events

### ISSUE THREAD

**Issue by fredfortier** [SitePen/dgrid#1110]
Is there a recommended way to emit custom events? I have created a List mixin to incorporate interesting business logic. I have tried this.emit() and got a function undefined error. In this particular scenario, I am looking to modify the behavior of the select events.

**Comment by fredfortier** [comment]
I think that I got it. Looking at the dgrid/Selection source code, I see that the grid is using dojo/on to emit against the "contentNode" element, not by extending dojo/Evented as I originally assumed. I could either override _fireSelectionEvent to emit my own select events. Or, I can emit a second type of select event again "contentNode". Since I do not know the internal side effects of removing the original select event, I may go with the latter. Please let me know if you have any thoughts around this.

**Comment by kfranqueiro** [comment]
dgrid's custom events all use on.emit to emit through the DOM.  Which node they emit off of depends on the particular event, which is why there isn't a central emit method (for example, `dgrid-sort` emits off of whatever element in the header was clicked).

What in particular are you trying to do with selection events?  Just add more information to events, or emit under different circumstances?

**Comment by fredfortier** [comment]
I want to fire business logic on single selection. In particular, fetch a full representation of the data element corresponding to the selection.

Since this logic is fairly expensive, essentially a server round-trip, I want to make sure that the user deliberately selected the item. Specifically, I do not want to fire this logic many times when the user is simply moving quickly through the grid using arrow keys.

What I did is subscribe to "dgrid-select" and "drgrid-deselect". I added my timeout and checks to ensure that the selection is deliberate. I emit a custom "single-select" event against the contentNode element. That way, I do not mess with the original events, and I get an event more specialized to my use case.
